File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        
        for(int i=0; i<sizes.length; i++){
            Arrays.sort(sizes[i]);
        }
        
        int n = 0, m = 0;
        
        for(int i=0; i<sizes.length; i++){
            n = Math.max(n, sizes[i][0]);
            m = Math.max(m, sizes[i][1]);
        }
        
        answer = n * m;
        
        return answer;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.kt
class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        var answer: Int = 0
        
        val sortedArray = sizes.map { it.sortedArray()}
        
        var maxCol = 0
        var maxRow = 0
        
        for(i in 0 until sortedArray.size){
            if(sortedArray[i][0] > maxCol){
                maxCol = sortedArray[i][0]
            }
            if(sortedArray[i][1] > maxRow){
                maxRow = sortedArray[i][1]
            }
        }
        
        answer = maxCol * maxRow
        
        
        return answer
    }
}